spec/controllers/trainees/confirm_reinstatements_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe Trainees::ConfirmReinstatementsController do
+  include ActiveJob::TestHelper
+
+  let(:current_user) { create(:user) }
+  let(:trainee) { create(:trainee, :deferred, trn: trn, provider: current_user.provider) }
+
+  before do
+    allow(controller).to receive(:current_user).and_return(current_user)
+  end
+
+  describe "#update" do
+    context "with a trainee with a trn" do
+      let(:trn) { "trn" }
+
+      it "transitions the trainee beack to trn_received" do
+        post :update, params: { trainee_id: trainee }
+        expect(trainee.reload).to be_trn_received
+      end
```

Comment:
Really just `beack`

```suggestion
      it "transitions the trainee back to trn_received" do
        expect {
          post :update, params: { trainee_id: trainee }
          trainee.reload
        }.to change {trainee.state} .from(:deferred) .to(:trn_received)
      end
```
but since `state` will change from `a` to `z`, maybe?